src/docker/mod.rs

```diff
@@ -81,17 +80,51 @@ impl<I: Image> Node<I> {
         let (to_write_in_env, client_endpoint, create_options) =
             <Node<I>>::write_env_file(&name, &mut create_options);
 
+        Self::get_image().and_then(move |_| {
+            Self::start_container(
+                envfile_path,
+                create_options,
+                client_endpoint,
+                to_write_in_env,
+            )
+        })
+    }
+
+    fn get_image() -> impl Future<Item = (), Error = shiplift::Error> {
         Docker::new()
             .images()
-            .pull(&PullOptions::builder().image(I::IMAGE).build())
-            .collect()
-            .and_then(move |_| {
-                Self::start_container(
-                    envfile_path,
-                    create_options,
-                    client_endpoint,
-                    to_write_in_env,
-                )
+            .list(&Default::default())
+            .map(|images| {
+                images
+                    .iter()
+                    .find(|image| {
+                        image
+                            .repo_tags
+                            .as_ref()
+                            .and_then(|repo_tags| {
+                                repo_tags
+                                    .iter()
+                                    .find(|tag| **tag == I::IMAGE.to_string())
+                                    .map(|_| true)
+                            })
+                            .unwrap_or(false)
+                    })
+                    .map(|_| true)
+                    .unwrap_or(false)
```

Comment:
![image](https://user-images.githubusercontent.com/5486389/66103117-0b8dc000-e5f8-11e9-815c-00c706574808.png)

Reply:
You just wanted to do a multi-line comment, didn't you? 😝